src/executor/values.rs

```diff
@@ -17,17 +19,35 @@ impl ValuesExecutor {
     #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
     pub async fn execute(self) {
         for chunk in self.values.chunks(PROCESSING_WINDOW_SIZE) {
+            type Type = DataTypeKind;
             // Create array builders.
-            let mut builders = self
-                .column_types
+            let column_types = &self.column_types;
+            let mut builders = column_types
                 .iter()
                 .map(|ty| ArrayBuilderImpl::with_capacity(chunk.len(), ty))
                 .collect_vec();
             // Push value into the builder.
             let dummy = DataChunk::single(0);
             for row in chunk {
-                for (expr, builder) in row.iter().zip_eq(&mut builders) {
+                for (expr, column_type, builder) in izip!(row, column_types, &mut builders) {
                     let value = expr.eval(&dummy)?;
+                    if let Type::Varchar(Some(size)) = column_type.kind {
```

Comment:
We can firstly store `Option<usize>` in a variable, and then check it. Therefore, we can write the `if item_length > size` part only once.